src/typescript-service.ts

```diff
@@ -1607,6 +1614,10 @@ export class TypeScriptService {
         await this.projectManager.ensureReferencedFiles(uri).toPromise()
         this.projectManager.didSave(uri)
     }
+    
+    public publishDiagnostics() {
```

Comment:
Does this method need to be public? In case you make it private: it looks like all private methods in this project should have a name that starts with `_`.